server/src/main/java/org/apache/druid/metadata/SQLMetadataStorageActionHandler.java

```diff
@@ -679,4 +962,81 @@ public Long getLockId(String entryId, LockType lock)
                             .findAny()
                             .orElse(null);
   }
+
+  private List<TaskMetadata> fetchTaskMetadatas(String tableName, String id, int limit)
+  {
+    List<TaskMetadata> taskMetadatas = new ArrayList<>();
+    connector.retryWithHandle(
+        new HandleCallback<Void>()
+        {
+          @Override
+          public Void withHandle(Handle handle)
+          {
+            String sql = StringUtils.format(
+                "SELECT * FROM %1$s WHERE id > '%2$s' AND type IS null ORDER BY id %3$s",
+                tableName,
+                id,
+                connector.limitClause(limit)
+            );
+            Query<Map<String, Object>> query = handle.createQuery(sql);
+            taskMetadatas.addAll(query.map(taskMetadataMapper).list());
+            return null;
+          }
+        }
+    );
+    return taskMetadatas;
+  }
+
+  private void updateTaskMetadatas(String tasksTable, List<TaskMetadata> taskMetadatas)
+  {
+    connector.retryWithHandle(
+        new HandleCallback<Void>()
+        {
+          @Override
+          public Void withHandle(Handle handle)
+          {
+            Batch batch = handle.createBatch();
+            String sql = "UPDATE %1$s SET type = '%2$s', group_id = '%3$s' WHERE id = '%4$s'";
+            for (TaskMetadata metadata : taskMetadatas) {
+              batch.add(StringUtils.format(sql, tasksTable, metadata.getType(), metadata.getGroupId(), metadata.getId())
+              );
+            }
+            batch.execute();
+            return null;
+          }
+        }
+    );
+  }
+
+
+  @Override
+  public boolean migrateTaskTable(String tableName)
+  {
+    log.info("Populate fields task and group_id of task entry table [%s] from payload", tableName);
+    String id = "";
+    int limit = 100;
+    while (true) {
+      List<TaskMetadata> taskMetadatas;
+      try {
+        taskMetadatas = fetchTaskMetadatas(tableName, id, limit);
+      }
+      catch (Exception e) {
+        log.warn(e, "Task migration failed while reading entries from task table");
+        return false;
+      }
+      if (taskMetadatas.isEmpty()) {
+        break;
+      }
+      try {
+        updateTaskMetadatas(tableName, taskMetadatas);
+      }
+      catch (Exception e) {
+        log.warn(e, "Task migration failed while updating entries in task table");
+        return false;
+      }
+      id = taskMetadatas.get(taskMetadatas.size() - 1).getId();
```

Comment:
In every iteration of the loop, it might be useful to add a debug log which prints the id upto which processing has been done so far.

Reply:
Added an info log indicating the total number of tasks migrated after each iteration. Also added a sleep of 1 second